packages/text-field/icon/index.tsx

```diff
@@ -102,23 +98,28 @@ export default class Icon extends React.Component<
         return '';
       },
       setAttr: (attr: keyof IconState, value: string | number) => (
-        this.setState((prevState) => ({...prevState, [attr]: value}))
+        this.setState((prevState) => {
+          if (attr === 'tabindex') {
+            value = Number(value);
+          }
+          return {...prevState, [attr]: value};
+        })
       ),
       removeAttr: (attr: keyof IconState) => (
         this.setState((prevState) => ({...prevState, [attr]: null}))
       ),
-      notifyIconAction: () => ( this.props.onSelect
-        ? this.props.onSelect()
-        : null
-      ),
+      setContent: () => undefined,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      notifyIconAction: () => (this.props.onSelect ? this.props.onSelect() : null),
```

Comment:
just do `notifyIconAction: this.props.onSelect!`, and add a default prop for onSelect: `onSelect: () => {}`

Reply:
I reflected it, but one question!
Is there any way to prevent type error for defaultProps?

Reply:
Gotcha - it needs to be `undefined`.